src/master-playlist-controller.js

```diff
@@ -273,9 +273,26 @@ export class MasterPlaylistController extends videojs.EventTarget {
         // playlist's `segments` list is already available, a media playlist won't be
         // requested, and loadedplaylist won't fire again, so the playlist handler must be
         // called on its own here.
-        if (this.sourceType_ === 'vhs-json' && this.initialMedia_.segments) {
-          this.handleUpdatedMediaPlaylist(this.initialMedia_);
+        if (this.sourceType_ !== 'vhs-json' && !this.initialMedia_.segments) {
+          return;
         }
+        updatedPlaylist = this.initialMedia_;
+      }
+
+      const codecs = updatedPlaylist.attributes.CODECS;
+      const segments = updatedPlaylist.segments || [];
+
+      // For non-fmp4 segments we also need to blacklist based on what the
+      // muxer supports. As non-fmp4 goes through the muxer
+      // but we have to wait for a playlist to download before
+      // we can know if it is fmp4 or not, so we can only really blacklist just
+      // before a playlist would get choosen
+      if (codecs && !segments[0].map && !muxerSupportsCodec(codecs)) {
```

Comment:
I wonder if we should just leave this as is for now. I think that scenario will only effect mixed ts/fmp4 playlists, right? 

Reasons:
Doing this now will make this pull request much more complex and we already rely on very simplistic blacklisting anyway.

For instance this is what I think we would have to do:
We will have to add a handler to `mediaSegmentRequest` that is called with enough segment data to determine the type of segment we are dealing with. Then we will have to check if the browser supports that codec if it is fmp4 or check if the muxer will support that codec if it is ts. Finally we will have to send an event up to the master playlist loader to blacklist this playlist and then abort loading of the remainder of the media segment based on that.

Reply:
The scenario can also be for TS playlists alone, see https://github.com/videojs/http-streaming/issues/151

Part of TBA was aiming to address the issue of assuming map meant fmp4. Though that has to be fully tested to see if it was fixed everywhere. 

If we want to go ahead with the PR without doing the more complicated approach, we can add a TODO. I think the case is rare enough that it might not be too big of an issue.

Reply:
I will add a TODO and work on it after this pull request.